OpenRA.Game/Network/Connection.cs

```diff
@@ -103,7 +103,7 @@ public sealed class NetworkConnection : IConnection
 
 		readonly List<byte[]> queuedSyncPackets = new List<byte[]>();
 		readonly Queue<(int Frame, int SyncHash, ulong DefeatState)> sentSync = new Queue<(int, int, ulong)>();
-		readonly Queue<(int Frame, OrderPacket Orders)> sentOrders = new Queue<(int, OrderPacket)>();
+		readonly Queue<(int Frame, Order[] Orders)> sentOrders = new Queue<(int, Order[])>();
```

Comment:
With these changes we need to make some extra steps, by transforming between Order[] and  OrderPacket. First we create a new instance of OrderPacket in `IConnection.Send` and then we do that again in `IConnection.Receive`. We should try to think of a cleaner solution for this.

Reply:
Maybe something like an OrderPacket.Combine that takes N inputs and returns one out?

Reply:
Putting the `Order[]` into the `OrderPacket` is cheap, it doesn't actually copy anything. Collecting all orders into a single array is needed in `IConnection.Receive` in some way anyway which is where the only copy happens.

I can add that `OrderPacket.Combine` if you prefer, but it shouldn't improve performance in any way (it actually adds an additional dereference in the receive path). Maybe JIT does something better, but that's hard to predict.

Reply:
My issue was not with any performance implications, but with keeping the code clean and as readable as possible.